src/App.test.js

```diff
@@ -1,8 +1,179 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import { render, screen, cleanup } from '@testing-library/react';
+import renderer from 'react-test-renderer';
+import userEvent from '@testing-library/user-event';
+import Calculator from './components/Calculator';
+import Home from './routes-component/Home';
+import Quote from './routes-component/Quotes';
+import operate from './logic/operate';
+import calculate from './logic/calculate';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+afterEach(() => {
+  cleanup();
+});
+
+describe('All components', () => {
+  it('should render Home component', () => {
+    render(<Home />);
+    const userElement = screen.getByTestId('user-id');
+    expect(userElement).toBeInTheDocument();
+    expect(userElement).toHaveTextContent('History');
+  });
+
+  it('should render Calculator component', () => {
+    render(<Calculator />);
+    screen.getAllByRole('button');
+  });
+
+  it('should match HOME snapshot', () => {
+    const home = renderer.create(<Home />).toJSON();
+    expect(home).toMatchSnapshot();
+  });
+
+  it('should match QUOTE snapshot', () => {
+    const quote = renderer.create(<Quote />).toJSON();
+    expect(quote).toMatchSnapshot();
+  });
+});
+
+describe('Conduct simple operations', () => {
+  const number1 = 9;
+  const number2 = 3;
+
+  it('Click number to display', () => {
+    render(<Calculator />);
+    const number = screen.getByTestId('cal-answer');
+    userEvent.click(screen.getByText('2'));
+    expect(number.innerHTML).toMatch('2');
+  });
+
+  const addition = operate(number1, number2, '+');
+  it('should perform an addition', () => {
+    expect(addition).toBe('12');
+  });
+
+  const subtraction = operate(number2, number1, '-');
+  it('should perform a subtraction', () => {
+    expect(subtraction).toBe('-6');
+  });
+
+  const multiplication = operate(number1, number2, 'x');
+  it('should perform a multiplication', () => {
+    expect(multiplication).toBe('27');
+  });
+
+  const division = operate(number1, number2, '÷');
+  it('should perform a division', () => {
+    expect(division).toBe('3');
+  });
+
+  const cantDivide = operate(number2, 0, '÷');
+  it('should check for no divisible by 0', () => {
+    expect(cantDivide).toBe("Can't divide by 0.");
+  });
+
+  const modulusOperator = operate(number1, number2, '%');
+  it('should check for modulus of a number', () => {
+    expect(modulusOperator).toBe('0');
+  });
+});
```

Comment:
- [ ] You did very well writing this test. But it can be better arranged according to the lesson on AAA. Here's how you can reorganize the given test using the Arrange-Act-Assert (AAA) pattern:

```javascript
describe('Conduct simple operations', () => {
  const number1 = 9;
  const number2 = 3;
  let addition, subtraction, multiplication, division, cantDivide, modulusOperator;

  beforeEach(() => {
    addition = operate(number1, number2, '+');
    subtraction = operate(number2, number1, '-');
    multiplication = operate(number1, number2, 'x');
    division = operate(number1, number2, '÷');
    cantDivide = operate(number2, 0, '÷');
    modulusOperator = operate(number1, number2, '%');
  });

  it('should click number to display', () => {
    render(<Calculator />);
    const number = screen.getByTestId('cal-answer');
    userEvent.click(screen.getByText('2'));
    expect(number.innerHTML).toMatch('2');
  });

  it('should perform an addition', () => {
    expect(addition).toBe('12');
  });

  it('should perform a subtraction', () => {
    expect(subtraction).toBe('-6');
  });

  it('should perform a multiplication', () => {
    expect(multiplication).toBe('27');
  });

  it('should perform a division', () => {
    expect(division).toBe('3');
  });

  it('should check for not dividing by 0', () => {
    expect(cantDivide).toBe("Can't divide by 0.");
  });

  it('should check for modulus of a number', () => {
    expect(modulusOperator).toBe('0');
  });
});
```

In this reorganized version, the Arrange-Act-Assert pattern is followed:

- **Arrange**: The setup and data preparation are done in the `beforeEach` hook, creating variables for the results of different operations.

- **Act**: Each individual test case (it-block) performs an action, such as clicking a number or calling the `operate` function.

- **Assert**: The expected outcomes are asserted using `expect` statements within each test case.

This organization makes the test cases more modular, easier to read, and follows a clear structure for setting up the test environment, performing actions, and verifying results.


- [ ] This test should be placed on a separate file and named according to the purpose of the test. A suitable filename for the test suite could be `calculatorOperations.test.js`. This filename reflects the purpose of the test suite, which is to test various operations of a calculator component.